### PR TITLE
Setup testing suite graphql

### PR DESCRIPTION
**Pull request by djtorel**
# Description

Implements testing suite fully configured to test GraphQL resolvers and queries

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

- [x] can test GraphQL resolvers
- [x] can test GraphQL queries

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,8 +3,20 @@ import express from 'express';
 import mongoose from 'mongoose';
 
 import { DATABASE_URL, PORT } from './util/env';
-import { typeDefs } from './typeDefs';
-import { resolvers } from './resolvers';
+import { typeDefs } from './graphql/typeDefs';
```
Comment (ckopecky) [inline src/index.js:6]: can you talk to me a little bit about the import for the env vars here? It is my experience that we would use 'dotenv' package and then require it at the top of the file. and then we would call the env vars using 'process.env.NAME_OF_VAR'

File: src/index.js [hunk 2]
```diff
@@ -3,8 +3,20 @@ import express from 'express';
...
+import { resolvers } from './graphql/resolvers';
+import SpotifyAPI from './graphql/datsources/spotify';
+
+// Setup dataSources our resolvers need
+const dataSources = () => ({
+  SpotifyAPI,
+});
+
+// Function that sets up global context for resolvers.
```
Comment (ckopecky) [inline src/index.js:18]: thank you for the comments. This helps.

File: src/index.js (hunk 1 above)
Comment (djtorel) [reply src/index.js:6]: It was a hack I did earlier to get it deployed. I'll implement dotEnv now and make another commit.

File: src/index.js (hunk 2 above)
Comment (CampbellD84) [reply src/index.js:18]: Comments really helped in seeing what is going on.

**Review by CampbellD84 — APPROVED** [review]
I see you added dotenv to set env variables and started to build out the Spotify API integration. Nice!

**Pull request merged**